pkg/ddl/syncer/syncer.go

```diff
@@ -368,14 +467,150 @@ func (s *schemaVersionSyncer) OwnerCheckAllVersions(ctx context.Context, jobID i
 				}
 				updatedMap[string(kv.Key)] = ""
 			}
+
+			if succ {
+				return nil
+			}
+			time.Sleep(checkVersInterval)
+			notMatchVerCnt++
+		}
+	}
+}
+
+// SyncJobSchemaVerLoop implements SchemaSyncer.SyncJobSchemaVerLoop interface.
+// this loop is quite light-weighted, so it's run on every node, and we don't need
+// to check whether it's the owner.
+// TODO run it on owner only when we refactor how DDL starts it's loop in a way similar
+// to DXF.
+func (s *schemaVersionSyncer) SyncJobSchemaVerLoop(ctx context.Context) {
+	for {
+		s.syncJobSchemaVer(ctx)
+		logutil.DDLLogger().Info("schema version sync loop interrupted, retrying...")
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(time.Second):
+		}
+	}
+}
+
+func (s *schemaVersionSyncer) syncJobSchemaVer(ctx context.Context) {
+	resp, err := s.etcdCli.Get(ctx, s.jobNodeVerPrefix, clientv3.WithPrefix())
+	if err != nil {
+		logutil.DDLLogger().Info("get all job versions failed", zap.Error(err))
+		return
+	}
+	s.mu.Lock()
+	// item might be saved and accessed outside, so we only clear data here, will
+	// clean up emptyAndNotUsed items later.
+	for _, item := range s.jobNodeVersions {
+		item.clearData()
+	}
+	s.mu.Unlock()
+	for _, oneKV := range resp.Kvs {
+		s.handleJobSchemaVerKV(oneKV, mvccpb.PUT)
+	}
+	s.mu.Lock()
+	// we might miss some DELETE events during retry, some items might be emptyAndNotUsed, remove them.
```

Comment:
every `resp` of line 498 is a snapshot, we can always remove all the old data and apply the `resp`, no need to have two removing

Reply:
https://github.com/pingcap/tidb/pull/53217/commits/738f11ca262bc7ca5e7472b1e8a561cc6adc1bf4